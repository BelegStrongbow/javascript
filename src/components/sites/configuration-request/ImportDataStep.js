import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";
import { defineMessages, FormattedMessage } from "react-intl";
import { LargeButton, makeButtonFullWidth, LargeSecondaryButton } from "../../Button";


let messages = defineMessages( {
	backupRequired: {
		id: "requestConfiguration.importDataCheck",
		defaultMessage: "Should we import data from another SEO plugin? We'll do this at no extra cost!" +
		" If yes, please select one of the available pluginsi n the dropdown from which you would like" +
		" to import data.",
	},
	backupExists: {
		id: "requestConfiguration.importDataLabel",
		defaultMessage: "Import from:",
	},
} );

const StyledLabel = styled.label`
	margin-left: 12px;
`;

const ButtonsContainer = styled.p`
	> button:not(:first-child) {
		margin-left: 12px;
	}
`;

const WideLargeButton = makeButtonFullWidth( LargeButton );
const WideSecondaryButton = makeButtonFullWidth( LargeSecondaryButton );

class ImportDataStep extends React.Component {
	constructor( props ) {
		super( props );

		this.state = {
			createBackup: props.createBackup,
		};

		this.handleInput = this.handleInput.bind( this );
		this.handleContinue = this.handleContinue.bind( this );
	}

	handleInput( event ) {
		this.setState( {
			createBackup: event.target.value === "true",
		} );
	}

	handleContinue() {
		if ( this.state.createBackup === null ) {
			return;
		}

		this.props.onSubmit( this.state );

		this.props.completeStep();
	}

	render() {
		return (
			<div>
				<p>
					<FormattedMessage
						id={ messages.backupRequired.id }
						defaultMessage={ messages.backupRequired.defaultMessage } />
				</p>
				<p>
					<input
						id="backupExists"
						type="radio"
						onChange={ this.handleInput }
						checked={ this.state.createBackup === false }
						value="false" />
					<StyledLabel htmlFor="backupExists">
						<FormattedMessage
							id={ messages.backupExists.id }
							defaultMessage={ messages.backupExists.defaultMessage } />
					</StyledLabel>
				</p>
				<p>
					<input
						id="backupMissing"
						type="radio"
						onChange={ this.handleInput }
						checked={ this.state.createBackup === true }
						value="true" />
					<StyledLabel htmlFor="backupMissing">
						<FormattedMessage
							id={ messages.backupMissing.id }
							defaultMessage={ messages.backupMissing.defaultMessage } />
					</StyledLabel>
				</p>
				<ButtonsContainer>
					<WideSecondaryButton onClick={ this.props.goToPreviousStep } >
						<FormattedMessage id="requestConfiguration.close" defaultMessage="back"/>
					</WideSecondaryButton>
					<WideLargeButton
						onClick={ this.handleContinue }
						type="submit"
						enabledStyle={ this.state.createBackup !== null }
						aria-label="continue"
					>
						<FormattedMessage id="requestConfiguration.continue" defaultMessage="continue"/>
					</WideLargeButton>
				</ButtonsContainer>
			</div>
		);
	}
}

ImportDataStep.propTypes = {
	createBackup: PropTypes.bool,
	onSubmit: PropTypes.func.isRequired,
	goToPreviousStep: PropTypes.func,
	completeStep: PropTypes.func,
};

export default ImportDataStep;
