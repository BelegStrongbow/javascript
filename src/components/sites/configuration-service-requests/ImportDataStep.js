import React from "react";
import styled from "styled-components";
import PropTypes from "prop-types";
import { defineMessages, FormattedMessage } from "react-intl";
import { LargeButton, makeButtonFullWidth, LargeSecondaryButton } from "../../Button";
import YoastSelect from "../../general/YoastSelect";
import ButtonsContainer from "../../general/ButtonsContainer";
import { StyledLabel } from "../../Labels";

let messages = defineMessages( {
	importDataRequired: {
		id: "requestConfiguration.importDataCheck",
		defaultMessage: "Should we import data from another SEO plugin? We'll do this at no extra cost!" +
		" If yes, please select one of the available plugins in the dropdown from which you would like" +
		" to import data.",
	},
	importDataExists: {
		id: "requestConfiguration.importDataLabel",
		defaultMessage: "Import from:",
	},
	noPluginSelected: {
		id: "requestConfiguration.noPluginSelected",
		defaultMessage: "No plugin selected",
	},
} );

const pluginOptions = [
	{ value: "none", label: messages.noPluginSelected.defaultMessage },
	{ value: "headspace2", label: "HeadSpace2" },
	{ value: "all-in-one_seo", label: "All-in-One SEO" },
	{ value: "jetpack_seo", label: "JetPack SEO" },
	{ value: "woothemes_seo_framework", label: "WooThemes SEO framework" },
];

const WideLargeButton = makeButtonFullWidth( LargeButton );
const WideSecondaryButton = makeButtonFullWidth( LargeSecondaryButton );

const TopSpaceButtonContainer = styled( ButtonsContainer )`
	padding-top: 16px;
`;

class ImportDataStep extends React.Component {

	constructor( props ) {
		super( props );
		this.state = {
			importFrom: props.importFrom || pluginOptions[ 0 ].value,
		};

		this.handleInput = this.handleInput.bind( this );
		this.handleContinue = this.handleContinue.bind( this );
	}
	/**
	 * Handles selected option of the selector in the import data step, by holding the selected option in the state.
	 *
	 * @param {object} event The event handling the input from the backup step.
	 *
	 * @returns {void}
	 */
	handleInput( event ) {
		this.setState( {
			importFrom: event.value,
		} );
	}
	/**
	 * Handles the continue button, when an option in the plugin selector is selected, the corresponding step is completed,
	 * and holds the user input in the props.
	 *
	 * @returns {void}
	 */
	handleContinue() {
		if ( this.state.importFrom === null ) {
			return;
		}

		this.props.onSubmit( this.state );
		this.props.completeStep();
	}

	render() {
		let value = this.state.importFrom;
		return (
			<div>
				<p>
					<FormattedMessage
						id={ messages.importDataRequired.id }
						defaultMessage={ messages.importDataRequired.defaultMessage } />
				</p>

				<StyledLabel htmlFor="import-data-exists">
					<FormattedMessage
						id={ messages.importDataExists.id }
						defaultMessage={ messages.importDataExists.defaultMessage } />
				</StyledLabel>

				<YoastSelect
					name="form-field-name"
					id="import-data-exists"
					value={ value }
					onChange={ this.handleInput }
					options={ pluginOptions }
					clearable={ false }
				/>
				<TopSpaceButtonContainer>
					<WideSecondaryButton onClick={ this.props.onBack } >
						<FormattedMessage id="requestConfiguration.close" defaultMessage="back"/>
					</WideSecondaryButton>
					<WideLargeButton
						onClick={ this.handleContinue }
						type="submit"
						enabledStyle={ this.state.importFrom !== null }
					>
						<FormattedMessage id="requestConfiguration.continue" defaultMessage="continue"/>
					</WideLargeButton>
				</TopSpaceButtonContainer>
			</div>
		);
	}
}

ImportDataStep.propTypes = {
	importFrom: PropTypes.string,
	onSubmit: PropTypes.func.isRequired,
	onBack: PropTypes.func,
	completeStep: PropTypes.func,
};

export default ImportDataStep;
