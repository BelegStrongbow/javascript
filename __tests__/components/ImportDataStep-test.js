import React from 'react';
import ImportDataStep from "../../src/components/ImportDataStep";

test('The ImportDataStep component matches the snapshot', () => {
	const component = <ImportDataStep
		importData={ 0 }
		onSubmit={ () => { console.log( "clicked on Continue" ) } }
		onBack={ () => { console.log( "clicked on Back" ); } }
		completeStep={ () => { console.log( "clicked on Continue" ); } } />;

	let tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});
