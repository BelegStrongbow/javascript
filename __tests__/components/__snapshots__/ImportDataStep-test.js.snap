// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The ImportDataStep component matches the snapshot 1`] = `
.c6 {
  height: 48px;
  margin: 0;
  padding: 0 16px;
  border: 0;
  border-bottom: 2px solid rgba(0,0,0,0.2);
  background-color: #64a60a;
  color: #fff;
  box-shadow: 0 2px 4px 0 rgba(0,0,0,0.2);
  border-radius: 4px;
  font: 400 14px/24px "Open Sans",sans-serif;
  text-transform: uppercase;
  cursor: pointer;
  vertical-align: top;
  -webkit-transition: background-color 150ms ease-out;
  transition: background-color 150ms ease-out;
}

.c6:hover,
.c6:focus {
  box-shadow: 0 4px 10px 0 rgba(0,0,0,0.2),inset 0 0 0 100px rgba( 0,0,0,0.1 );
}

.c6:active {
  -webkit-transform: translateY( 1px );
  -ms-transform: translateY( 1px );
  transform: translateY( 1px );
  box-shadow: none;
}

.c8 {
  text-shadow: 0px 0px 2px #000;
}

.c5 {
  color: #000;
  box-shadow: none;
  background-color: #fff;
  border: 1px solid #000;
  min-width: 152px;
}

.c5:hover,
.c5:focus {
  background-color: #64a60a;
  color: #fff;
  box-shadow: 0 2px 4px 0 rgba(0,0,0,0.2);
  border: 0;
  text-shadow: 0px 0px 2px #000;
}

.c7 {
  min-width: 152px;
}

.c1 {
  width: 100%;
  box-shadow: 0 2px 4px 0 rgba(0,0,0,0.2);
}

.c1.is-focused:not(.is-open) > .Select-control {
  border-color: #64a60a;
  box-shadow: none;
  background: #fff;
}

.c1 .Select-control {
  height: 48px;
  border-radius: 0;
}

.c1 .Select-placeholder {
  line-height: 48px;
}

.c1.Select--single > .Select-control .Select-value {
  line-height: 48px;
}

.c1 .Select-value-label {
  line-height: 48px;
}

.c1 .Select-input {
  line-height: 48px;
}

.c1 .Select-option {
  color: #000;
}

.c1 .Select-option.is-selected {
  background-color: #6dff003d;
}

.c1 .Select-option.is-focused {
  background-color: #64a60a;
  color: #fff;
}

.c3 {
  -webkit-flex: 1 0 200px;
  -ms-flex: 1 0 200px;
  flex: 1 0 200px;
  padding-bottom: 16px;
  text-align: right;
}

.c3 a,
.c3 button {
  margin-left: 12px;
}

.c2 {
  padding-top: 16px;
}

.c0 {
  display: inline-block;
  font-size: 1em;
  margin: 0 0 8px;
}

@media screen and ( max-width:800px ) {
  .c4 {
    width: 100%;
    background-image: none;
    padding: 12px 16px;
  }
}

@media screen and (max-width:800px) {
  .c3 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
    -webkit-box-pack: center;
    -webkit-justify-content: center;
    -ms-flex-pack: center;
    justify-content: center;
  }

  .c3 a,
  .c3 button {
    margin-left: 0;
    margin-bottom: 8px;
  }
}

<div>
  <p>
    <span>
      Should we import data from another SEO plugin? We'll do this at no extra cost! If yes, please select one of the available plugins in the dropdown from which you would like to import data.
    </span>
  </p>
  <label
    className="c0"
    htmlFor="import-data-exists"
  >
    <span>
      Import from:
    </span>
  </label>
  <div
    className="Select c1 has-value is-searchable Select--single"
    style={undefined}
  >
    <input
      disabled={false}
      name="form-field-name"
      type="hidden"
      value="none"
    />
    <div
      className="Select-control"
      onKeyDown={[Function]}
      onMouseDown={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      style={undefined}
    >
      <span
        className="Select-multi-value-wrapper"
        id="react-select-2--value"
      >
        <div
          className="Select-value"
          style={undefined}
          title={undefined}
        >
          <span
            aria-selected="true"
            className="Select-value-label"
            id="react-select-2--value-item"
            role="option"
          >
            No plugin selected
          </span>
        </div>
        <div
          className="Select-input"
          style={
            Object {
              "display": "inline-block",
            }
          }
        >
          <input
            aria-activedescendant="react-select-2--value"
            aria-describedby={undefined}
            aria-expanded="false"
            aria-haspopup="false"
            aria-label={undefined}
            aria-labelledby={undefined}
            aria-owns=""
            className={undefined}
            id="import-data-exists"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            required={false}
            role="combobox"
            style={
              Object {
                "boxSizing": "content-box",
                "width": "5px",
              }
            }
            tabIndex={undefined}
            value=""
          />
          <div
            style={
              Object {
                "height": 0,
                "left": 0,
                "overflow": "scroll",
                "position": "absolute",
                "top": 0,
                "visibility": "hidden",
                "whiteSpace": "pre",
              }
            }
          >
            
          </div>
        </div>
      </span>
      <span
        className="Select-arrow-zone"
        onMouseDown={[Function]}
      >
        <span
          className="Select-arrow"
          onMouseDown={[Function]}
        />
      </span>
    </div>
  </div>
  <div
    className="c2 c3"
  >
    <button
      className="c4 c5 c6"
      onClick={[Function]}
      type="button"
    >
      <span>
        back
      </span>
    </button>
    <button
      className="c4 c7 Button-kDSBcD c8 c6"
      onClick={[Function]}
      type="submit"
    >
      <span>
        continue
      </span>
    </button>
  </div>
</div>
`;
